pkg/kritis/cryptolib/verifier.go

```diff
@@ -80,23 +86,32 @@ type verifier struct {
 	pkixVerifier
 	pgpVerifier
 	jwtVerifier
+	authenticatedAttFormer
 	authenticatedAuthChecker
 }
 
 // NewVerifier creates a Verifier interface for verifying Attestations.
-// `imageDigest` contains the digest of the image that was signed over. This
-// should be provided directly by the policy evaluator, NOT by the Attestation.
+// `image` contains the untruncated image name <image_name@digest> of the image
+// that was signed. This should be provided directly by the policy evaluator,
+// NOT by the Attestation.
 // `publicKeySet` contains a list of PublicKeys that the Verifier will use to
 // try to verify an Attestation.
-func NewVerifier(imageDigest string, publicKeySet []PublicKey) (Verifier, error) {
+func NewVerifier(image string, publicKeySet []PublicKey) (Verifier, error) {
+	digest, err := name.NewDigest(image, name.StrictValidation)
```

Comment:
I think disecting `image` string here should also be avoided here because it's also part of "user operation".

Reply:
We discussed offline that in this case it is okay to dissect `image` here, as we can provide a failure message directly to the user if the `image` is invalid